### ISSUE TITLE
Total Noob

### ISSUE THREAD

**Issue by dwterrell** [hyperbrew/bolt-cep#14]
Hello Hyper Brew-

I'm a noob programmer.  I've downloaded your bolt-cep on my iMac and I was wondering if there is a short beginner's guide you can direct me to.  I've unzipped the file, but every time I try to install npm, I get dependencies errors for the RollUp dependency.  Has anyone else reported this or do most people know workarounds?  Am I using a wrong version of npm??? I'm just really not sure where to find the answers.  Could you point me in the right direction?  Thank you.

<img width="573" alt="Screen Shot 2022-02-28 at 4 29 21 PM" src="https://user-images.githubusercontent.com/100632325/156061748-463c14ee-0d8a-4667-a954-5c96eaf92ffe.png">
<img width="208" alt="Screen Shot 2022-02-28 at 4 28 53 PM" src="https://user-images.githubusercontent.com/100632325/156061750-14e60586-99f4-4cda-9661-e388af521883.png">

**Comment by justintaylor-dev** [comment]
Hi there! If you're brand new to programming, I wouldn't recommend starting with this boilerplate as trying to learn JavaScript / HTML / CSS and React all at once is a lot.

I'd recommend starting with Scripting if you're using an app that supports ScriptUI like After Effects / Photoshop / or Illustrator, or Extensions if you're using Premiere or any non-ScriptUI compatible app.

Here's some helpful posts:

[](https://hyperbrew.co/blog/scripts-extensions-plugins/)
[](https://hyperbrew.co/blog/building-adobe-scripts/)
[](https://hyperbrew.co/blog/building-adobe-extensions/)

--------------

If you come back to this once you're comfortable with scripting and extensions in Adobe Apps, here's what I would recommend to resolve your issue.

- Install yarn: [](https://classic.yarnpkg.com/en/docs/install#windows-stable)
- Clone the repo with `git clone https://github.com/hyperbrew/bolt-cep.git`
- cd into the repo
- Install dependencies with `yarn`
- Build with `yarn build`
- and develop with `yarn dev`

**Comment by dwterrell** [comment]
Hey Justin-

Thank you for the response.  I really appreciate.

I may have labelled my topic incorrectly, I've made several AE scripts and now I'm working on a Premiere extension.  I have made interfaces with resource strings and recently made an interface for my Premiere extension with CEP. It's just a simple interface for a slate maker script.

I have it working with a base CEP extension template from https://premiereonscript.com, but I was trying to take my skills to another (better, more concise) level.  I have watched most of the Davide's course (https://www.ps-scripting.com/uxp-react.html) so I'm a bit familiar with some of the aspects of React.  And I can make my way around HTML and CSS, well enough for my needs at the moment.

I am not too familiar with Terminal commands and fixing errors though.  I've been going over some YouTube tutorials and Stackoverflow articles  to figure those out (still struggling though).

Thank you for your next steps plans.  Once again, I appreciate your response.

<img width="413" alt="Screen Shot 2022-03-01 at 10 05 51 AM" src="https://user-images.githubusercontent.com/100632325/156193767-9585872a-d4d1-45a8-82ab-b81620fae08e.png">
